python/cudf_polars/cudf_polars/callback.py

```diff
@@ -52,19 +148,29 @@ def execute_with_cudf(
     nt
         NodeTraverser
 
-    raise_on_fail
-        Should conversion raise an exception rather than continuing
-        without setting a callback.
+    config
+        GPUEngine configuration object
 
     exception
         Optional exception, or tuple of exceptions, to catch during
         translation. Defaults to ``Exception``.
 
     The NodeTraverser is mutated if the libcudf executor can handle the plan.
     """
+    device = config.device
+    memory_resource = config.memory_resource
+    raise_on_fail = config.config.get("raise_on_fail", False)
```

Comment:
Can we raise if an user passes an invalid kwarg?

Reply:
Well, from the point of view of the polars docs, `raise_on_fail` is an invalid kwarg, so yes, I suppose, but what do we gain?

Reply:
Ah I forgot that the kwargs is an internal API for us.

Could be still useful to check to protect against typos (e.g. yesterday I used "mr" instead of "memory_resource" which silently does nothing), but I won't push this further.

Reply:
Raise now for unsupported kwargs.